### PR TITLE
feat(neon): Support data URIs in NeonUrlImage

### PR DESCRIPTION
**Pull request by provokateurin**
Apps can also return inline SVGs as icons, so we need to support them as well.

### CODE REVIEW

File: packages/neon/neon/lib/src/widgets/image.dart [hunk 1]
```diff
@@ -354,19 +354,27 @@ class NeonUrlImage extends StatelessWidget {
   Widget build(final BuildContext context) {
     final account = this.account ?? NeonProvider.of<AccountsBloc>(context).activeAccount.value!;
 
+    final uri = Uri.parse(url);
+    final dataUri = uri.scheme == 'data' ? UriData.fromUri(uri) : null;
```
Comment (Leptopoda) [inline packages/neon/neon/lib/src/widgets/image.dart:358]:
```suggestion
    final dataUri = uri.data;
```
I prefer inlining this in the below check as this variable would be redundant.
Also for your information `uri.isScheme('data')` would be preferred as it does the comparison case insensitive.

**Review by Leptopoda — CHANGES_REQUESTED** [review]

File: packages/neon/neon/lib/src/widgets/image.dart (hunk 1 above)
Comment (provokateurin) [reply packages/neon/neon/lib/src/widgets/image.dart:358]: The data URI is also used in the hint, that's why I moved it outside.

Comment (Leptopoda) [reply packages/neon/neon/lib/src/widgets/image.dart]: Any specific reason you don't like `uri.data` or just so the code doesn't look ugly.
I agree that there isn't any difference.

**Review by Leptopoda — APPROVED** [review]

File: packages/neon/neon/lib/src/widgets/image.dart (hunk 1 above)
Comment (provokateurin) [reply packages/neon/neon/lib/src/widgets/image.dart]: Oh I didn't know there was a getter for it. I'll use it instead

**Pull request merged**